components/add_new_cv_form.tsx

```diff
@@ -40,21 +86,37 @@ export default function AddNewCvForm({ id }: Props) {
   }
 
   async function upsert(values: any) {
+    const CV = { ...values };
+    delete CV.projects;
+
     if (id) {
-      return supabase.from("cv").update(values).eq("id", id);
+      return supabase.from("cv").update(CV).eq("id", id).select();
     }
 
     const {
       data: { session },
     } = await supabase.auth.getSession();
-    return supabase.from("cv").insert({
-      ...values,
-      created_by: session?.user.id,
-    });
+
+    return supabase
+      .from("cv")
+      .insert({
+        ...CV,
+        created_by: session?.user.id,
+      })
+      .select();
+  }
+
+  async function upsertProjects(values: any, cvData: any) {
+    if (values.id) {
+      return supabase.from("projects").update(values).eq("id", values.id);
+    }
+    return supabase.from("projects").insert({ ...values, cv_id: cvData[0].id });
```

Comment:
There's already an upsert function in supabase, no need for this if https://supabase.com/docs/reference/javascript/upsert